classes/class-wc-connect-taxjar-integration.php

```diff
@@ -713,34 +825,35 @@ public function create_or_update_tax_rate( $line_item_key, $location, $rate, $ta
 		} else {
 			// Insert a rate if we did not find one
 			$this->_log( ':: Adding New Tax Rate ::' );
+			$this->_log( $tax_rate );
 			$rate_id = WC_Tax::_insert_tax_rate( $tax_rate );
 			WC_Tax::_update_tax_rate_postcodes( $rate_id, wc_clean( $location['to_zip'] ) );
 			WC_Tax::_update_tax_rate_cities( $rate_id, wc_clean( $location['to_city'] ) );
 		}
 
-		$this->_log( 'Tax Rate ID Set for ' . $line_item_key . ': ' . $rate_id );
-		$this->rate_ids[ $line_item_key ] = $rate_id;
+		$this->_log( 'Tax Rate ID Set for ' . $rate_id );
+		return $rate_id;
 	}
 
 	/**
 	 * Wrap SmartCalcs API requests in a transient-based caching layer.
 	 *
-	 * Modified from TaxJar's plugin (removed use of TLC Transients)
-	 * See: https://github.com/taxjar/taxjar-woocommerce-plugin/blob/82bf7c58/includes/class-wc-taxjar-integration.php#L463
+	 * Unchanged from the TaxJar plugin.
+	 * See: https://github.com/taxjar/taxjar-woocommerce-plugin/blob/4b481f5/includes/class-wc-taxjar-integration.php#L451
 	 *
 	 * @param $json
 	 *
 	 * @return mixed|WP_Error
 	 */
 	public function smartcalcs_cache_request( $json ) {
-		$cache_key = 'wcs_tax_' . hash( 'md5', $json );
+		$cache_key = 'tj_tax_' . hash( 'md5', $json );
 		$response  = get_transient( $cache_key );
 
 		if ( false === $response ) {
 			$response = $this->smartcalcs_request( $json );
 
 			if ( 200 == wp_remote_retrieve_response_code( $response ) ) {
-				set_transient( $cache_key, $response, HOUR_IN_SECONDS );
+				set_transient( $cache_key, $response, $this->cache_time );
```

Comment:
Why differ from the TaxJar plugin here?

Reply:
We were differing *before*, now it's exactly the same as the Taxjar plugin.

Reply:
Whoops! I missed that. 👍